cylc/flow/parsec/fileparse.py

```diff
@@ -239,27 +265,59 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
         flines = inline(
             flines, fdir, fpath, False, viewcfg=viewcfg, for_edit=asedit)
 
+    template_vars['CYLC_VERSION'] = __version__
+
     # process with EmPy
     if do_empy:
+        if (
+            extra_vars['templating detected'] == 'empy:suite.rc' and
+            not re.match(r'^#![Ee]m[Pp]y\s*', flines[0])
+        ):
+            if not re.match(r'^#!', flines[0]):
+                flines.insert(0, '#!empy')
+            else:
+                raise FileParseError(
+                    "Your file has shebang line incompatible with empy"
+                    " variables."
+                )
         if flines and re.match(r'^#![Ee]m[Pp]y\s*', flines[0]):
             LOG.debug('Processing with EmPy')
+            tvars = copy(template_vars)
+            if extra_vars['templating detected'] == 'empy:suite.rc':
+                for key, value in extra_vars['template variables'].items():
+                    tvars[key] = value
             try:
                 from cylc.flow.parsec.empysupport import empyprocess
             except (ImportError, ModuleNotFoundError):
                 raise ParsecError('EmPy Python package must be installed '
                                   'to process file: ' + fpath)
-            flines = empyprocess(flines, fdir, template_vars)
+            flines = empyprocess(flines, fdir, tvars)
 
     # process with Jinja2
     if do_jinja2:
+        if (
+            extra_vars['templating detected'] == 'jinja2:suite.rc' and
+            not re.match(r'^#![jJ]inja2\s*', flines[0])
+        ):
+            if not re.match(r'^#!', flines[0]):
+                flines.insert(0, '#!jinja2')
+            else:
+                raise FileParseError(
+                    "Your file has shebang line incompatible with jinja2"
+                    " variables."
+                )
         if flines and re.match(r'^#![jJ]inja2\s*', flines[0]):
             LOG.debug('Processing with Jinja2')
+            tvars = copy(template_vars)
+            if extra_vars['templating detected'] == 'jinja2:suite.rc':
+                for key, value in extra_vars['template variables'].items():
+                    tvars[key] = value
```

Comment:
Are rose vars supposed to override command line suite variables or should this be the other way around?

Reply:
CLI should override.

Reply:
thus?
```suggestion
            if extra_vars['templating detected'] == 'jinja2:suite.rc':
                for key, value in template_vars.items():
                    extra_vars[key] = value
            tvars = extra_vars
```

Reply:
I think that I've fixed this @TomekTrzeciak .
Question for you and @oliver-sanders  : If we use `cylc <app> --set=VAR=99` and in the `rose-suite.conf`: `VAR=144` then var will have a different variable in the `ROSE_SUITE_VARS` because these are set in the plugin. It potentially allowsu workflow designers to prevent users using `--set` by writing `ROSE_SUITE_VARS['VAR']` in the suite. Is this ok?

Reply:
Fine.